packages/flutter/lib/src/material/slider.dart

```diff
@@ -534,9 +543,16 @@ class _SliderState extends State<Slider> with TickerProviderStateMixin {
     super.dispose();
   }
 
+  double _preciseLerp(double value){
+    return double.parse(value.toStringAsFixed(widget.precision!));
+  }
+
   void _handleChanged(double value) {
     assert(widget.onChanged != null);
-    final double lerpValue = _lerp(value);
+    double lerpValue = _lerp(value);
+    if (widget.precision != null) {
+      lerpValue = _preciseLerp(lerpValue);
+    }
```

Comment:
What if precision is null? will we still face the same issue?

I suggest to fix the issue directly instead of introduce a new property. FYI, this was already working without any additional property so it still should be posible to fix the issue without any additional property.

Maybe just round the resulting value?

Reply:
Hi @pedromassango , Thanks for the review!


> Maybe just round the resulting value?

<details>
<summary>Example</summary>

```
Slider(
	value: value,
	label: '$value',
	min: 0,
	max: 10,
	divisions: 20,
	onChanged: (v) => setState(() {
		value = v;
		print(value);
		}),
 	),
```

</details>

User wants 20 divisions of 10 that means 0.5 as a single division. If we round the value it end up giving us values 0, 1.0, 2.0 



> What if precision is null? will we still face the same issue?

Thanks for pointing out this. We could set ```precision``` to 0 (Which is equivalent to ```round```) as a default. and modify assertion to   
```
assert(precision != null && (precision >= 0 && precision <= 20))

```
This approach may affect many users as it is a **breaking change**. (That's the only reason I made it null by default).

### Why I chosen to add ```precision``` property workaround?

 instead of rounding it up internally to certain precision to 2,3, or foo. Cause we never know what exact ```precision``` user wants. So I tried to give it a bit customizable approach.

If I'm wrong anywhere please point it out to me. 

Meanwhile I try to figure out some other workarounds to fix this issue without introducing new property.

Thanks!

Reply:
(sorry, I wrote the above comment months ago, I just hadn't submitted it)